### DISCUSSION TITLE
How to use with Google Fonts

### DISCUSSION THREAD

**Discussion by MatteoGauthier** [vercel/next.js#21486] [Help] [answered]
I wanted to use Next.js with a Google fonts, optimized during the build

**Comment by darrylmorley** [comment, marked as answer]
You can set up a _document.js file in pages and use the below code.

```
import Document, { Html, Head, Main, NextScript } from 'next/document'

class MyDocument extends Document {
  render() {
    return (
      <Html>
        <Head>
          <link rel="icon" href="/favicon.ico" />
          <link rel="preconnect" href="https://fonts.gstatic.com" />
          <link href="https://fonts.googleapis.com/css2?family=Inter:wght@400;600;700&display=swap" rel="stylesheet" />
        </Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    )
  }
}

export default MyDocument
```

Obviously replacing the font for your preferred. Documentation can be found here: https://nextjs.org/docs/advanced-features/custom-document

**Reply by MatteoGauthier** [reply]
Thanks a lot for your fast reply, i understand more clearly now thks.

**Comment by andrzejchmura** [comment]
Highly recommend @leerob's blog post on this topic: https://leerob.io/blog/fonts.

Spoiler: you might want to self-host a font downloaded from Google Fonts for better performance.

**Reply by MatteoGauthier** [reply]
Very good article, everyone should see it 😉
